src/System.Linq/src/System/Linq/Reverse.cs

```diff
@@ -91,16 +91,7 @@ public override void Dispose()
             public TSource[] ToArray()
             {
                 TSource[] array = _source.ToArray();
-
-                // Array.Reverse() involves boxing for non-primitive value types, but
-                // checking that has its own cost, so just use this approach for all types.
-                for (int i = 0, j = array.Length - 1; i < j; ++i, --j)
-                {
-                    TSource temp = array[i];
-                    array[i] = array[j];
-                    array[j] = temp;
-                }
-
+                Array.Reverse<TSource>(array);
```

Comment:
Normally I hate explicit type parameters when they aren't needed, but it seems you were correct here as the CI error wouldn't have happened and desktop might have ended up getting the less efficient behaviour you want to avoid.